scripts/updateVersion.ts

```diff
@@ -0,0 +1,116 @@
+const nuspecVersionRegex = /<version>(.*)\$prereleaseSuffix\$<\/version>/i;
+const constantsPath = path.join(repoRoot, 'src', 'constants.ts');
+const constantsVersionRegex = /version = '(.*)'/i;
+
+const args = parseArgs(process.argv.slice(2));
+if (args.validate) {
+    validateVersion();
+} else if (args.version) {
+    updateVersion(args.version);
+} else {
+    console.log(`This script can be used to either update the version of the worker or validate that the repo is in a valid state with regards to versioning.
+
+NOTE: For the types package, only the major & minor version need to match the worker. We follow the same pattern as DefinitelyTyped as described here:
+https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library
+
+Example usage:
+
+npm run updateVersion -- --version 3.3.0
+
+npm run updateVersion -- --validate`);
+    throw new Error('Invalid arguments');
+}
+
+function validateVersion() {
+    const packageJson = readJSONSync(packageJsonPath);
+    const packageJsonVersion = packageJson.version;
+
+    const typesPackageJson = readJSONSync(typesPackageJsonPath);
+    const typesPackageJsonVersion = typesPackageJson.version;
+
+    const nuspecVersion = getVersion(nuspecPath, nuspecVersionRegex);
+
+    const constantsVersion = getVersion(constantsPath, constantsVersionRegex);
+
+    console.log('Found the following versions:');
+    console.log(`- package.json: ${packageJsonVersion}`);
+    console.log(`- types/package.json: ${typesPackageJsonVersion}`);
+    console.log(`- Worker.nuspec: ${nuspecVersion}`);
+    console.log(`- src/constants.ts: ${constantsVersion}`);
+
+    const parsedVersion = semver.parse(packageJsonVersion);
+    const parsedTypesVersion = semver.parse(typesPackageJsonVersion);
+
+    if (
+        !packageJsonVersion ||
+        !nuspecVersion ||
+        !constantsVersion ||
+        !typesPackageJsonVersion ||
+        !parsedVersion ||
+        !parsedTypesVersion
+    ) {
+        throw new Error('Failed to detect valid versions in all expected files');
+    } else if (nuspecVersion !== packageJsonVersion || constantsVersion !== packageJsonVersion) {
+        throw new Error(`Worker versions do not match.`);
+    } else if (parsedVersion.major !== parsedTypesVersion.major || parsedVersion.minor !== parsedTypesVersion.minor) {
+        throw new Error(`Types package does not match the major/minor version of the worker.`);
+    } else {
+        console.log('Versions match! 🎉');
+    }
+}
+
+function getVersion(filePath: string, regex: RegExp): string {
+    const fileContents = readFileSync(filePath).toString();
+    const match = fileContents.match(regex);
+    if (!match) {
+        throw new Error(`Failed to find match for "${regex.source}".`);
+    }
+    return match[1];
+}
+
+function updateVersion(newVersion: string) {
+    updatePackageJsonVersion(repoRoot, newVersion);
+
+    if (newVersion.endsWith('.0')) {
```

Comment:
What if `newVersion` is something like `4.0`?

Reply:
Well, `4.0` isn't valid semver https://semver.org/

Reply:
Well, do we check that `newVersion` is a valid semver? 😛

Regardless, let's say, for whatever reason, we were upgrading from `3.0.0` to `3.1.1`, don't we still want to update the types package?

Reply:
I suppose we don't check locally, but `validateVersion` does make sure its valid semver, so the build would fail as soon as someone put up a PR.

Going from `3.0.0` to `3.1.1` is also an invalid semver pattern. Per the docs: "Patch version MUST be reset to 0 when minor version is incremented." If you have a simple suggestion to check that edge case I will add it, but otherwise I don't think it's worth the time for an engineering script. Hopefully if someone did that it would be caught in code review.

Reply:
Fair enough. The only thing I had in mind was to get the old version first and compare with the new version, but I suppose that might not be worth it as you say :)